Cabal/src/Distribution/Simple/BuildTarget.hs

```diff
@@ -320,7 +321,9 @@ resolveBuildTarget pkg userTarget fexists =
   where
     classifyMatchErrors errs
       | Just expected' <- NE.nonEmpty expected
-                            = let (things, got:|_) = NE.unzip expected' in
+                            = let unzip' = fmap fst &&& fmap snd
+                                  -- ^ to work around deprecated NE.unzip
```

Comment:
Well, by "comment" I mean an explanation that we can't use `Data.Functor.unzip` for compatibility reasons. The deprecation is eminent and sooner than we might perceive no one will remember of the generic `NE.unzip`. So, I'm not sure it's worth to put a comment about it.

Maybe, actually putting all the comments in the commit message is enough. Nothing will help too much with reading this code, and all the history belongs to commit messages imo.

Reply:
I'm happy to oblige. I'll force-push this branch with a single commit that explains why `Data.Functor.unzip` is unsuitable for the time being. Eventually `NE.unzip` may be reinstated, but the performance gain from that change will likely be minuscule.

Reply:
Great, thanks!